manim/mobject/types/vectorized_mobject.py

```diff
@@ -2117,7 +2127,44 @@ def construct(self):
                         (gr-circle_red).animate.shift(RIGHT)
                     )
         """
-        return super().add(*vmobjects)
+
+        def get_type_error_message(invalid_obj, invalid_i):
+            return (
+                f"Only values of type {vmobject_render_type.__name__} can be added "
+                f"as submobjects of VGroup, but the value "
+                f"{repr(invalid_obj)} (at index {invalid_i}) is of type "
+                f"{type(invalid_obj).__name__}."
+            )
+
+        vmobject_render_type = (
+            OpenGLVMobject if config.renderer == RendererType.OPENGL else VMobject
+        )
+        valid_vmobjects = []
+
+        for vmobject_i, vmobject in enumerate(vmobjects):
+            if isinstance(vmobject, vmobject_render_type):
+                valid_vmobjects.append(vmobject)
+            elif isinstance(vmobject, Iterable) and not isinstance(
+                vmobject, (Mobject, OpenGLMobject)
+            ):
+                for subvmobject_i, subvmobject in enumerate(vmobject):
+                    if not isinstance(subvmobject, vmobject_render_type):
+                        raise TypeError(
+                            get_type_error_message(subvmobject, subvmobject_i)
+                        )
+                    valid_vmobjects.append(subvmobject)
+            elif isinstance(vmobject, Iterable) and isinstance(
+                vmobject, (Mobject, OpenGLMobject)
+            ):
+                # This is if vmobject is an empty Mobject
```

Comment:
Is this comment true? At first glance it seems like a non-empty `Group` might pass this check.

Reply:
I've removed this, but let me know if you want me to add more inline comments